examples/java/src/main/java/org/apache/beam/examples/complete/TfIdf.java

```diff
@@ -179,9 +173,11 @@ public ReadDocuments(Iterable<URI> uris) {
           uriString = uri.toString();
         }
 
-        PCollection<KV<URI, String>> oneUriToLines = pipeline
-            .apply("TextIO.Read(" + uriString + ")", TextIO.read().from(uriString))
-            .apply("WithKeys(" + uriString + ")", WithKeys.<URI, String>of(uri));
+        PCollection<KV<URI, String>> oneUriToLines =
+            pipeline
+                .apply("TextIO.Read(" + uriString + ")", TextIO.read().from(uriString))
+                .apply("WithKeys(" + uriString + ")", WithKeys.<URI, String>of(uri))
+                .setCoder(KvCoder.of(StringDelegateCoder.of(URI.class), StringUtf8Coder.of()));
```

Comment:
Yeah, I was thinking that the next step would be to augment ParDo, Map/FlatMapElements, and WithKeys to allow them to take a coder (coders, in case of multi-output pardo - they could go into TupleTagList perhaps). Not sure if there's a way around that.